### PR TITLE
Task/insert upsert transaction

### PR DESCRIPTION
**Pull request by IvanHdzC**
This PR creates a transaction to upsert-insert data. 

Also, changes the statement of the insert to build a prepared statement. Notice that the new insert is only performed with the upsert transaction. If the last_data function is disabled, then the isert is performed as usual.

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackendImpl.java [hunk 1]
```diff
@@ -516,38 +541,91 @@ public void createErrorTable(String destination)
...
+            upsertPreparedStatement = SQLQueryUtils.addJsonValues(upsertStatement, lastData, attrNativeTypes);
+            upsertPreparedStatement.executeBatch();
+
+            connection.commit();
+            LOGGER.info("Finished transaction: \n" + upsertPreparedStatement + "\n Inserted ROWS: " + insertedRows.length + " " + upsertQuery);
+
         } catch (SQLTimeoutException e) {
-            throw new CygnusPersistenceError(sqlInstance.toUpperCase() + " Data insertion error. Query: `" + preparedStatement, "SQLTimeoutException", e.getMessage());
+            cygnusSQLRollback(connection);
+            throw new CygnusPersistenceError(sqlInstance.toUpperCase() + " " + e.getNextException() + " Data insertion error. Query: `" + connection, "SQLTimeoutException", e.getMessage());
```
Comment (AlvaroVega) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackendImpl.java:611]: upsertQuery should be printed, not only  connection

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackendImpl.java [hunk 2]
```diff
@@ -516,38 +541,91 @@ public void createErrorTable(String destination)
...
         } catch (SQLException e) {
-            throw new CygnusBadContextData(sqlInstance.toUpperCase() + " Data insertion error. Query: `" + preparedStatement, "SQLException", e.getMessage());
+            cygnusSQLRollback(connection);
+            throw new CygnusBadContextData(sqlInstance.toUpperCase() + " " + e.getNextException() + " Data insertion error. Query: `" + connection, "SQLException", e.getMessage());
```
Comment (AlvaroVega) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackendImpl.java:614]: the same upper comment: upsertQuery should be printed, not only connection

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**